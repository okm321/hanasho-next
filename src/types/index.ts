export * from './header'
export * from './product'
export * from './slide'
