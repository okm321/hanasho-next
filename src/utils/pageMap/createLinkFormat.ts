import { PageMapCategory } from '@/types'
import { checkNeverType } from '../common'

/**
 * ページマップのリンクを作成する
 * @param category - ページマップのカテゴリ
 * @param id - ページマップのID
 */
export const createLinkFormat = (category: PageMapCategory, id: number) => {
  switch (category) {
    case PageMapCategory.Use:
      return `/search/${id}?type=use`
    case PageMapCategory.Style:
      return `/search/${id}?type=style`
    case PageMapCategory.Topic:
      return `/search/topic/${id}`
    case PageMapCategory.Menu:
      return '/'
  }
  checkNeverType(category)
}
