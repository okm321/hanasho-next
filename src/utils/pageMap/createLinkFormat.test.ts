import { PageMapCategory } from '@/types'
import { createLinkFormat } from './createLinkFormat'

describe('createLinkFormat', () => {
  it('用途でさがす場合', () => {
    expect(createLinkFormat(PageMapCategory.Use, 1)).toMatchSnapshot()
  })
  it('スタイルでさがす場合', () => {
    expect(createLinkFormat(PageMapCategory.Style, 1)).toMatchSnapshot()
  })
  it('特集でさがす場合', () => {
    expect(createLinkFormat(PageMapCategory.Topic, 1)).toMatchSnapshot()
  })
  it('メニューの場合', () => {
    expect(createLinkFormat(PageMapCategory.Menu, 1)).toMatchSnapshot()
  })
  it('例外の場合', () => {
    expect(createLinkFormat(9 as PageMapCategory, 1)).toMatchSnapshot()
  })
})
