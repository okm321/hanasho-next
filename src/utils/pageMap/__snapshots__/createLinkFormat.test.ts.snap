// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createLinkFormat スタイルでさがす場合 1`] = `"/search/1?type=style"`;

exports[`createLinkFormat メニューの場合 1`] = `"/"`;

exports[`createLinkFormat 例外の場合 1`] = `undefined`;

exports[`createLinkFormat 特集でさがす場合 1`] = `"/search/topic/1"`;

exports[`createLinkFormat 用途でさがす場合 1`] = `"/search/1?type=use"`;
